packages/client/hmi-client/src/temp/tera-simulation-workflow.vue

```diff
@@ -1,22 +1,86 @@
 <template>
-	<infinite-canvas debug-mode :last-transform="lastTransform" @save-transform="saveTransform">
+	<infinite-canvas
+		debug-mode
+		@contextmenu="toggleContextMenu"
+		:last-transform="canvasTransform"
+		@save-transform="saveTransform"
+	>
 		<template #foreground></template>
 		<template #data>
-			<div style="font-size: 24px; padding: 10px; background: #9ef">This is a DataLayer DIV</div>
+			<ContextMenu ref="contextMenu" :model="contextMenuItems" />
+
+			<ul v-for="node in nodes">
+				<tera-workflow-node :node="node" :initialTransform="canvasTransform"></tera-workflow-node>
+			</ul>
 		</template>
 	</infinite-canvas>
 </template>
 
 <script setup lang="ts">
+import { ref } from 'vue';
 import InfiniteCanvas from '@/components/widgets/tera-infinite-canvas.vue';
+import { Operation, WorkflowNode, WorkflowStatus } from '@/types/workflow';
+import TeraWorkflowNode from './tera-workflow-node.vue';
+import ContextMenu from 'primevue/contextmenu';
+
+const nodes = ref<WorkflowNode[]>([]);
+const contextMenu = ref();
+const newNodePosition = ref<{ x: number; y: number }>({ x: 0, y: 0 });
+const canvasTransform = ref<d3.ZoomTransform>();
```

Comment:
Is wrapping `d3.ZoomTransform` inside a ref necessary here? 

We have had a lot of problems in making 3rd party data structures reactive in the past, some times they become massive memory-hogs, and sometimes they cause call-stack explosions. It seems like we are just treating it as a normal variable so I think we can take out the `ref` part.